linux_docker_resources/Dockerfile

```diff
@@ -38,6 +38,9 @@ RUN apt-get update && apt-get install -y build-essential ccache cmake pkg-config
 # Install build and test dependencies of ROS 2 packages.
 RUN apt-get update && apt-get install -y clang-format-3.8 cppcheck git pydocstyle pyflakes python3-coverage python3-flake8 python3-mock python3-nose python3-pep8 uncrustify
 
+# Install build dependencies for turtlebot demo.
+RUN if test -n ${INSTALL_TURTLEBOT2_DEMO_DEPS}; apt-get update && apt-get install -y libeigen3-dev libtinyxml-dev
```

Comment:
I think you'll need to add this to the top of the dockerfile as with https://github.com/ros2/ci/commit/a5ff6c4e14dbe70d829676f5aeb533a15ff3bab2#diff-a7a7037d3453364950fd80f5f91ed7eeR3